test/e2e/density.go

```diff
@@ -253,6 +253,21 @@ var _ = Describe("Density [Skipped]", func() {
 			stop := make(chan struct{})
 			go controller.Run(stop)
 
+			// Create a listener for api updates
+			updateCount := 0
+			cache.NewReflector(
+				// Listen pod update from all nodes
+				cache.NewListWatchFromClient(c, "pods", api.NamespaceAll, fields.Everything()),
+				&api.Pod{},
+				cache.NewUndeltaStore(
+					func(objs []interface{}) {
```

Comment:
The UndeltaStore sends the entire list of pods according to https://github.com/kubernetes/kubernetes/blob/master/pkg/client/cache/undelta_store.go#L19. I think you should increment the `updateCount` by one whenever receiving an update.